tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/lhlo_legalize_to_affine.cc

```diff
@@ -98,6 +98,66 @@ struct DotOpConverter : public OpRewritePattern<DotOp> {
   }
 };
 
+struct ConcatOpConverter : public OpRewritePattern<ConcatenateOp> {
+  using OpRewritePattern<ConcatenateOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ConcatenateOp op,
+                                PatternRewriter& rewriter) const override {
+    Location loc = op.getLoc();
+    Value output = op.output();
+    MemRefType outputType = output.getType().cast<MemRefType>();
+    unsigned outputRank = outputType.getRank();
+    ArrayRef<int64_t> outputShape = outputType.getShape();
+
+    ValueRange operands = op.val();
+    uint64_t concatDim = op.dimension();
+    int64_t prevBound = 0;
+
+    for (Value operand : operands) {
+      MemRefType operandType = operand.getType().cast<MemRefType>();
+      ArrayRef<int64_t> operandShape = operandType.getShape();
+      if (!operandType.hasStaticShape()) return failure();
+      SmallVector<Value, 3> indices;
```

Comment:
Please move this declaration closer to its use.

Reply:
Done. Thanks.